### PR TITLE
Working php7 from fuck-php7

### PR DESCRIPTION
**Pull request by robske110**
# **_DO NOT MERGE THIS YET!_**

Bugs to be fixed: #480 
- Fix shutdown bugs (Server is being killed by ServerKiller thread, but not critical as everything is saved, only logger can't exit right)
- RakLib likes to crash
- ~~PlayerJoinCrashes~~
- ~~Entity class is all messed up~~
- ~~DoubleTagChestInventory~~

### CODE REVIEW

**Comment by inxomnyaa** [comment]
You can PR branchs into branchs? didnt knew that

File: src/pocketmine/MemoryManager.php [hunk 1]
```diff
@@ -32,16 +32,16 @@ class MemoryManager{
...
 	/** @var \WeakRef[] */
 	private $leakWatch = [];
 
 	private $leakInfo = [];
 
 	private $leakSeed = 0;
 
+	private $chunkCache;
+	private $cacheTrigger;
+
```
Comment (inxomnyaa) [inline src/pocketmine/MemoryManager.php:44]: why

File: src/pocketmine/plugin/PluginManager.php [hunk 2]
```diff
@@ -413,7 +410,7 @@ public function subscribeToPermission($permission, Permissible $permissible){
 		if(!isset($this->permSubs[$permission])){
 			$this->permSubs[$permission] = [];
 		}
-		$this->permSubs[$permission][spl_object_hash($permissible)] = new \WeakRef($permissible);
```
Comment (inxomnyaa) [inline src/pocketmine/plugin/PluginManager.php:416]: why no weakref

File: src/pocketmine/plugin/PluginManager.php [hunk 3]
```diff
@@ -486,6 +484,7 @@ public function getDefaultPermSubscriptions($op){
 		$subs = [];
 
 		if($op === true){
+			return $this->defSubsOp;
```
Comment (inxomnyaa) [inline src/pocketmine/plugin/PluginManager.php:487]: this will make the code ignore the next lines..

Comment (robske110) [inline src/pocketmine/plugin/PluginManager.php:487]: Yes, that was crashing

File: src/pocketmine/plugin/PluginManager.php (hunk 2 above)
Comment (robske110) [inline src/pocketmine/plugin/PluginManager.php:416]: Crashing

File: src/pocketmine/MemoryManager.php (hunk 1 above)
Comment (robske110) [inline src/pocketmine/MemoryManager.php:44]: I did a commit to remove the weakref stuff here and then reverted it the wrong way

**Pull request merged**